ipaserver/install/server/replicainstall.py

```diff
@@ -1456,11 +1462,41 @@ def init(installer):
         installer.server = None
     if installer.replica_file is None:
         installer.password = installer.host_password
+        # unless some credentials are given, we need to have at least
+        # an admin principal and its password so these are not asked for
+        # twice (first in client-install, then in conncheck)
+        # TODO: remove when client-install is performed as a function call
+        if (not installer.unattended and installer.password is None and
+                installer.keytab is None and
+                installer.admin_password is None and
+                installer._ccache is None):
+            if not installer.principal:
+                try:
+                    # get the principal interactively, can't be empty
+                    installer.principal = ipautil.user_input(
+                            "User authorized to enroll computers",
+                            allow_empty=False)
+                    root_logger.debug(
+                            "will use principal provided as option: %s",
+                            installer.principal)
+                except Exception as e:
+                    print()
+                    # higher-level error so script usage is not printed
+                    raise ScriptError(str(e))
+            # the principal is set now, we need its password
+            try:
+                installer.admin_password = getpass.getpass(
+                    "Password for {}: ".format(installer.principal))
+            except EOFError:
+                print()
+                installer.admin_password = None
+            if not installer.admin_password:
+                # higher-level error so script usage is not printed
+                raise ScriptError("Password must be provided for %s." %
+                                  installer.principal)
```

Comment:
This whole block should be moved to `promote_check()`, interactive prompts elsewhere are done in `*_check()` as well.